src/main/java/com/airdnb/clone/domain/stay/entity/AvailableAmenity.java

```diff
@@ -0,0 +1,56 @@
+package com.airdnb.clone.domain.stay.entity;
+
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.FetchType;
+import jakarta.persistence.ForeignKey;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.ManyToOne;
+import jakarta.persistence.Table;
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+
+@Getter
+@NoArgsConstructor(access = AccessLevel.PROTECTED)
+@AllArgsConstructor(access = AccessLevel.PROTECTED)
+@Builder
+@Table(name = "AVAILABLE_AMENITY")
+@Entity
+public class AvailableAmenity {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Column(name = "AVAILABLE_ID")
+    private Long id;
+
+    @ManyToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "STAY_ID", foreignKey = @ForeignKey(name = "FK_STAY_AVAILABLE_ID"))
+    private Stay stay;
+
+    @ManyToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "AMENITY_ID", foreignKey = @ForeignKey(name = "FK_AMENITY_AVAILABLE_ID"))
+    private Amenity amenity;
+
+    public AvailableAmenity setStay(Stay stay) {
+        if (this.stay != null) {
+            this.stay.getAmenities().remove(this);
```

Comment:
```suggestion
            this.stay.removeAmenity(this)
```

현재는 AvailableAmenity 가 Stay 의 내부정보를 알고 있는 코드처럼 보여요. 객체지향적으로 좋지 않아 보입니다.